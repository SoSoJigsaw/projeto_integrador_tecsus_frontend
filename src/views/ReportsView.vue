<template>
    <transition name="fade" mode="out-in">
        <div v-if="isVisible" class="container-principal">
            <h1>Relat√≥rios</h1>
        </div>
    </transition>
</template>

<script lang="ts">
export default {
  data() {
    return {
      isVisible: false
    };
  },
  mounted() {
    setTimeout(() => {
      this.isVisible = true;
    }, 1000);
  }
}
</script>


<style>
@import "../assets/base.css";

.container-principal {
    margin-left: 4%;
    margin-top: 3%;
    margin-right: 4%;
    margin-bottom: 3%;
}

.container-principal h1 {
    color: #776AE3;
    font-size: 6vmin;
    font-weight: 400;
}
</style>