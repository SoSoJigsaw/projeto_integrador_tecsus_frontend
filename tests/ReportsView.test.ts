import { describe, it, expect, vi } from 'vitest'
import { mount, flushPromises } from '@vue/test-utils';
import ReportsView from '../src/views/ReportsView.vue';

const createWrapper = (isVisible = true) => {
  return mount(ReportsView, {
    data() {
      return { isVisible };
    } 
  });
};

describe('ReportsView.vue', () => {

  it('should mount the component correctly', async () => {
    const wrapper = createWrapper();
    await flushPromises();
    expect(wrapper.exists()).toBe(true)
  })

  it('should start with isVisible set to false', async () => {
    const wrapper = createWrapper(false);
    await flushPromises();
    expect(wrapper.vm.isVisible).toBe(false)
  })

  it('should display the main container after 1 second', async () => {
    vi.useFakeTimers()
    const wrapper = createWrapper();
    vi.advanceTimersByTime(1000)
    await flushPromises();
    expect(wrapper.find('.container-principal').isVisible()).toBe(true)
    vi.useRealTimers()
  })

  it('should display the "Relatórios" title after becoming visible', async () => {
    vi.useFakeTimers()
    const wrapper = createWrapper();
    vi.advanceTimersByTime(1000)
    await flushPromises();
    expect(wrapper.find('.container-principal h1').text()).toContain('Relatórios')
    vi.useRealTimers()
  })

  it('should clean up when the component is destroyed', async () => {
    const wrapper = createWrapper();
    await flushPromises();
    wrapper.vm.destroyed = async function() {
      expect(wrapper.vm.isVisible).toBe(undefined)
    };
    
  })

  it('should render without errors with default props', async () => {
    const wrapper = createWrapper(false);
    await flushPromises();
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should always render the main container', async () => {
    const wrapper = createWrapper();
    await flushPromises();
    expect(wrapper.find('.container-principal').exists()).toBe(true)
  })

});