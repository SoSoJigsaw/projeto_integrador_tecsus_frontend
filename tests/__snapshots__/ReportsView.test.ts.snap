// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReportsView.vue > should render without errors with default props 1`] = `"<!--v-if-->"`;
